stan/math/rev/mat/fun/cholesky_decompose.hpp

```diff
@@ -265,16 +437,21 @@ inline Eigen::Matrix<var, -1, -1> cholesky_decompose(
       accum_i = accum;
     }
   } else {
-    cholesky_block* baseVari = new cholesky_block(A, L_A);
-    size_t pos = 0;
-    for (size_type j = 0; j < L.cols(); ++j) {
-      for (size_type i = j; i < L.cols(); ++i) {
-        L.coeffRef(i, j).vi_ = baseVari->variRefL_[pos++];
-      }
-      for (size_type k = 0; k < j; ++k)
-        L.coeffRef(k, j).vi_ = dummy;
+#ifdef STAN_OPENCL
+    if (L_A.rows()
+        > opencl_context.tuning_opts().cholesky_size_worth_transfer) {
+      cholesky_opencl* baseVari = new cholesky_opencl(A, L_A);
+      internal::set_lower_tri_coeff_ref(L, baseVari->variRefL_);
+    } else {
+      cholesky_block* baseVari = new cholesky_block(A, L_A);
+      internal::set_lower_tri_coeff_ref(L, baseVari->variRefL_);
     }
+#else
+    cholesky_block* baseVari = new cholesky_block(A, L_A);
+    internal::set_lower_tri_coeff_ref(L, baseVari->variRefL_);
```

Comment:
Seems like this line is the same in all 3 branches of the `ifdef` and `if` - can we put it after?

Reply:
It wont work, I tried it but I get the compiler error `error: ‘baseVari’ was not declared in this scope`.
The problem is that the first one is a `cholesky_opencl` class while the other 2 are `cholesky_block`.

Reply:
Yeah, Sean is there a trick for the above scoping issue idk about in C++? Agree it would be nice to not have this repeated multiple times

Reply:
Oops. The only trick I can think of would be if this was a virtual method call on an interface declared above the if. Might be a potential way to refactor if we saw lots of use cases, but for now we can leave it.